### PR TITLE
SNOW-1814176: Implement SeriesGroupBy.unique.

### PR DESCRIPTION
**Pull request by sfc-gh-mvashishtha**
Fixes SNOW-1814176

### CODE REVIEW

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py [hunk 1]
```diff
@@ -19418,3 +19418,146 @@ def timedelta_property(
...
+                + "single Snowpark column."
+            )
+            by_snowflake_quoted_identifiers_list.append(entry[0])
+
+        # There is no built-in snowflake function to aggregation unique values
+        # of a column into array while preserving a certain order. We implement
+        # the aggregation in the following steps:
+        # 1) Project a new column representing the row position of each row
+        #    within each combination of group + data column value.
+        # 3) Filter the result to the rows where the new column is equal to 1,
```
Comment (sfc-gh-jjiao) [inline src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:19508]: is there a step 2)?

**Review by sfc-gh-jjiao — APPROVED** [review]
LGTM with a comment

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py [hunk 2]
```diff
@@ -19418,3 +19418,146 @@ def timedelta_property(
...
+                "Internal error. Each grouping label should correspond to a "
+                + "single Snowpark column."
+            )
+            by_snowflake_quoted_identifiers_list.append(entry[0])
+
+        # There is no built-in snowflake function to aggregation unique values
+        # of a column into array while preserving a certain order. We implement
+        # the aggregation in the following steps:
+        # 1) Project a new column representing the row position of each row
+        #    within each combination of group + data column value.
```
Comment (sfc-gh-nkumar) [inline src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:19507]: Why we need group + data columns here? Should it be just grouping columns?

Comment (sfc-gh-mvashishtha) [reply src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:19507]: We need to keep the rows where the combination of group + data column value appears for the next time. I explained that in the next step. Let me know if I can make the wording clearer.

Comment (sfc-gh-nkumar) [reply src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:19507]: Ah, I understand it now. No this is fine. Thanks!

**Review by sfc-gh-nkumar — APPROVED** [review]

**Pull request merged**